src/ImageSharp/IO/IFileSystem.cs

```diff
@@ -9,16 +9,32 @@ namespace SixLabors.ImageSharp.IO;
 internal interface IFileSystem
 {
     /// <summary>
-    /// Returns a readable stream as defined by the path.
+    /// Opens a file as defined by the path and returns it as a readable stream.
     /// </summary>
     /// <param name="path">Path to the file to open.</param>
-    /// <returns>A stream representing the file to open.</returns>
+    /// <returns>A stream representing the opened file.</returns>
     Stream OpenRead(string path);
 
     /// <summary>
-    /// Creates or opens a file and returns it as a writable stream as defined by the path.
+    /// Opens a file as defined by the path and returns it as a readable stream
+    /// that can be used for asynchronous reading.
     /// </summary>
     /// <param name="path">Path to the file to open.</param>
-    /// <returns>A stream representing the file to open.</returns>
+    /// <returns>A stream representing the opened file.</returns>
+    Stream OpenReadAsynchronous(string path);
```

Comment:
@antonfirsov I didn't know whether you'd prefer adding separate methods like this or just having one method with a `FileOptions` argument (that way, you could potentially use other options like `FileOptions.SequentialScan` as well). I chose this approach only because this appears to be an abstraction over `System.IO` and having a `FileOptions` argument would leak the abstraction, but let me know if you'd prefer that approach regardless.

Reply:
I think since the interface is internal, I wouldn't want to expose implementation details. Users can already do use their own streams.